### PR TITLE
update enigma processing

### PR DESCRIPTION
**Pull request by zfisch**
update enigma processing to work with new steps outined in readme, remove old files no longer in use, include refgene and other enigma processing scripts in repo

There is a lot more that could/should be done for enigma processing to reduce the number of manual steps. The main issue is that the format of the data we receive continues to change, and requires very different handling (e.g. enigma data looks like it used to have clinvar accessions included, now it requires a new file to be merged in).

Submitting a pull request now due to time pressure (I need to continue working through other pipeline issues and ui code).

### CODE REVIEW

File: pipeline/enigma/README.md [hunk 1]
```diff
@@ -2,28 +2,27 @@
...
+3. Open enigma data in google spreadsheets, go to variants tab, select file -> download as -> Tab-separated values (tsv, .current sheet).
+4. Remove all text in tsv file from step 3 except for the column names and variant data.
+5. Rename column "Alternate designations" to "BIC Nomenclature", rename column "Official allele name" to "Abbrev AA change", remove last column "Replaces ClinVarAccessions".
+6. Remove all text in clinvar accession file before column (header) names.
+7. Run `python merge-clinvaraccessions.py -e /PATH/TO/ENIGMA_TSV_FILE_FROM_STEP_5 -c /PATH/TO/CLINVAR_ACCESSION_DATA_TXT_FILE_FROM_STEP_6 -o /PATH/TO/OUTPUT_FILE`
+8. Run `python enigma-processing.py -i /PATH/TO/OUTPUT_FILE_FROM_STEP_& -o /PATH/TO/OUTPUT_DIRECTORY_CONTAINING_PROCESSED_ENIGMA_FILES/ENIGMA_last_updated_MM_DD_YY_hg38.tsv -g /PATH/TO/hg38.fa`
+9. Make sure all available ENIGMA_last_updated_MM_DD_YY_hg38.tsv files are in the same directory, then run `python enigma-merge_hg38.py -i /PATH/TO/OUTPUT_DIRECTORY_CONTAINING_PROCESSED_ENIGMA_FILES/ -o /PATH/TO/OUTPUT_DIRECTORY/`
+10. This procedure produces an `ENIGMA_combined_hg38.tsv` file that can be used in the luigi pipeline.
```
Comment (melissacline) [inline pipeline/enigma/README.md]: @zfisch The one thing I don't see in this list is updating the preprocessed files on Synapse.  Let's keep doing that, as it's been useful to check Synapse to verify details of some of the ENIGMA variants.

File: pipeline/enigma/enigma-merge_hg38.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+def main():
+    default_input_files = sorted(glob.glob("output/*last_updated*hg38*tsv"))
+    parser = argparse.ArgumentParser()
+    parser.add_argument('-i', '--input_folder', help='Link to folder with all processed Enigma files to merge')
+    parser.add_argument('-o', '--output_folder', help='Folder to contain output file')
+    parser.add_argument('-f', '--input_files', nargs="*",
+                        help='Links to processed ENIGMA files to be merged if input folder is not provided.',
+                        default=default_input_files)
+    args = parser.parse_args()
```
Comment (melissacline) [inline pipeline/enigma/enigma-merge_hg38.py]: Moving forward, the next time we get new ENIGMA data, we could change this script to get the input files directly from Synapse.  That would save one manual step.

File: pipeline/enigma/enigma-processing.py [hunk 3]
```diff
@@ -115,7 +112,7 @@ def convert_OMIM_id(OMIM_id):
 
 def convert_HGVS(hgvs_c, GENOME):
     chrom, offset, ref, alt = pyhgvs.parse_hgvs_name(
-        hgvs_c, genome, get_transcript=get_transcript)
+        hgvs_c, GENOME, get_transcript=get_transcript)
```
Comment (melissacline) [inline pipeline/enigma/enigma-processing.py]: Oh man!  Was the genome argument undefined in the old code?

File: pipeline/enigma/enigma-processing.py [hunk 4]
```diff
@@ -133,9 +130,27 @@ def HGVS_cDNA_to_protein(hgvs_c):
     return hgvs_p
 
 
+def HGVS_p_to_AA_abrev(HGVS_p):
```
Comment (melissacline) [inline pipeline/enigma/enigma-processing.py]: Great!  This addresses issue #184 !  In enigma_postprocess.py, we can remove the function fixBrokenAminoAcidChange(), and simplify the code base that little bit more.

File: pipeline/enigma/README.md (hunk 1 above)
Comment (zfisch) [inline pipeline/enigma/README.md]: adding in a separate pr

File: pipeline/enigma/enigma-merge_hg38.py (hunk 2 above)
Comment (zfisch) [inline pipeline/enigma/enigma-merge_hg38.py]: 👍

**Pull request merged**